app/controllers/responses_controller.rb

```
@@ -0,0 +1,249 @@
+# frozen_string_literal: true
+
+class ResponsesController < ApplicationController
+  before_action :set_response, only: [:show, :update, :submit, :unsubmit, :destroy]
+
+  # Authorization: determines if current user can perform the action
+  def action_allowed?
+    case action_name
+    when 'create', 'show', 'index'
+      true
```

RuboCop (1.86.1)

[warning] Literal `true` used in void context.

(Lint/Void)